ark/internal/controller/tool_controller.go

```diff
@@ -47,5 +52,92 @@ func (r *ToolReconciler) updateToolStatus(ctx context.Context, tool *arkv1alpha1
 }
 
 func (r *ToolReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.Add(&builtinToolInitializer{client: mgr.GetClient()}); err != nil {
+		return fmt.Errorf("failed to add builtin tool initializer: %w", err)
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).For(&arkv1alpha1.Tool{}).Named("tool").Complete(r)
 }
+
+type builtinToolInitializer struct {
+	client client.Client
+}
+
+func (b *builtinToolInitializer) Start(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+	log.Info("initializing built-in tools")
+
+	if err := b.ensureBuiltinToolsExist(ctx); err != nil {
+		log.Error(err, "failed to create built-in tools")
+		return fmt.Errorf("failed to create built-in tools: %w", err)
+	}
+
+	log.Info("built-in tools initialization completed")
+	return nil
+}
+
+func (b *builtinToolInitializer) ensureBuiltinToolsExist(ctx context.Context) error {
+	builtinTools := []genai.ToolDefinition{
+		genai.GetNoopTool(),
+		genai.GetTerminateTool(),
+	}
+
+	for _, builtinTool := range builtinTools {
+		if err := b.ensureSingleBuiltinTool(ctx, builtinTool.Name, builtinTool.Description, builtinTool.Parameters); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *builtinToolInitializer) ensureSingleBuiltinTool(ctx context.Context, name, description string, parameters map[string]any) error {
+	log := logf.FromContext(ctx)
+	log.Info("ensuring built-in tool exists", "tool", name)
+
+	tool := &arkv1alpha1.Tool{}
+	key := client.ObjectKey{Name: name, Namespace: "default"}
+
+	err := b.client.Get(ctx, key, tool)
+	if err == nil {
+		log.Info("built-in tool already exists", "tool", name)
+		return nil
+	}
+
+	if !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to get tool %s: %w", name, err)
+	}
+
+	if err := b.createBuiltinTool(ctx, name, description, parameters); err != nil {
+		return fmt.Errorf("failed to create built-in tool %s: %w", name, err)
+	}
+
+	log.Info("created built-in tool", "tool", name)
+	return nil
+}
+
+func (b *builtinToolInitializer) createBuiltinTool(ctx context.Context, name, description string, parameters map[string]any) error {
+	parametersJSON, err := json.Marshal(parameters)
+	if err != nil {
+		return fmt.Errorf("failed to marshal parameters: %w", err)
+	}
+
+	tool := &arkv1alpha1.Tool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "default",
+			Labels: map[string]string{
+				"ark.mckinsey.com/builtin": "true",
```

Comment:
we have some annotation consts we might be able to use but do we need it given we have a tool type in the spec?